crates/net/discv4/src/lib.rs

```diff
@@ -135,6 +135,17 @@ impl Discv4 {
         Ok(discv4)
     }
 
+    /// Returns a new instance with the given channel directly
+    ///
+    /// NOTE: this is only intended for test setups.
+    #[cfg(feature = "mock")]
```

Comment:
usually, we hide this behind `test-utils` feature. may be worth keeping it consistent

Reply:
yep, good call.